### DISCUSSION TITLE
Course Related

### DISCUSSION THREAD

**Discussion by shreyaajainn31** [snigi-gupta/AttendingUniversityAtBuffalo#21] [Q&A] [answered]
I had some queries regarding the course selection for fall 2021 **MS in CS**: 

1. When does the registration of the courses start? If it is our first time for registration, do we register before entering the college or after?
2. How should we choose our courses, for example, if we want **2 Easy, 1 Medium, and 1 Hard** --or-- **2 Easy 2 Medium courses**, how do we decide for the first semester?
3. Should we study any specific topic before coming to college?

**Comment by snigi-gupta** [comment, marked as answer]
Hi @shreyaajainn31,

1- This was easily googable. [here is the link though.](https://registrar.buffalo.edu/registration/) On the left side the semesters are mentioned, you can choose the one you wish to see.

2- I already have all this mentioned in another readme.md file for CSE students (not sure if you have already gone through it). The second folder (folder name: Computer Science Important Tips) in the github repo (AttendingUniversityAtBuffalo) has all the information. If you still have doubts after reading that document, feel free to post your follow-up questions here, I will be happy to answer them for you 😄. I had also shared the link with you on LinkedIn when you reached out to me 😄. [here is the link though.](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips) I have shared what difficulty level course combination students should usually choose in their 1st, 2nd and 3rd semester.

3- Before coming to ub make sure you are brushing up your DSA (Data Structures and Algorithms), go through the basics on geeksforgeeks - like search-sort algorithms, read about arrays, linked-lists, stacks, queues, trees and graphs. Familiarize yourself with Python and practice 10-20 EASY leetcode or hackerrank questions, so that you have a hold of the language and syntax. If you do all this before arriving to ub, you should be in a decent place to keep up with the pace of the courses. If you do have time, feel free to do more. But I believe this is the bare minimum, you should consider working on.

I hope this helps! ❤️
